tensorflow/compiler/mlir/xla/ir/infer_fusibility_op_interface.td

```diff
@@ -0,0 +1,119 @@
+
+// OpInterface to query if an op is fusible and to query the shape equality
+// constraint among the inputs and outputs of an op.
+def InferFusibilityOpInterface : OpInterface<"InferFusibilityOpInterface"> {
+  let description = [{
+    Interface to query if an op is fusible and to query the shape equality
+    constraint among the inputs and outputs of an op.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/[{If true, this op can be fused with its operands
+      }],
+      /*retTy=*/"bool",
+      /*methodName=*/"isFusibleWithOperand",
+      /*args=*/(ins),
+      /*methodBody=*/[{}],
+      /*defaultImplementation=*/[{
+        /// Returns whether this op can be fused with its operands
+        return true;
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{If true, this op can be fused with its consumers
+      }],
+      /*retTy=*/"bool",
+      /*methodName=*/"isFusibleWithConsumer",
+      /*args=*/(ins),
+      /*methodBody=*/[{}],
+      /*defaultImplementation=*/[{
+        /// Return whether this op can be fused withh its consumers
+        return true;
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return whether two inputs have the same shape (assuming no"
+               "implicit broadcasting).",
+      /*retTy=*/"bool",
+      /*methodName=*/"inferInputsShapeEquality",
+      /*args=*/(ins "int":$lhs, "int":$rhs),
+      /*methodBody=*/[{}],
+      /*defaultImplementation=*/[{
+        /// Return whether two outputs have the same shape.
+        assert(lhs < op.getOperation()->getNumOperands() && lhs >= 0 &&
+               rhs < op.getOperation()->getNumOperands() && rhs >= 0);
+        return lhs == rhs;
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return whether two outputs have the same shape (assuming no"
+               " implicit broadcasting).",
+      /*retTy=*/"bool",
+      /*methodName=*/"inferOutputsShapeEquality",
+      /*args=*/(ins "int":$lhs, "int":$rhs),
+      /*methodBody=*/[{}],
+      /*defaultImplementation=*/[{
+        /// Return whether two outputs have the same shape.
+        assert(lhs < op.getOperation()->getNumResults() && lhs >= 0 &&
+               rhs < op.getOperation()->getNumResults() && rhs >= 0);
+        return lhs == rhs;
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return whether two the input and the output have the same"
+               " shape (assuming no implicit broadcasting).",
+      /*retTy=*/"bool",
+      /*methodName=*/"inferInputOutputShapeEquality",
+      /*args=*/(ins "int":$input, "int":$output),
+      /*methodBody=*/[{}],
+      /*defaultImplementation=*/[{
+        /// Return whether two outputs have the same shape.
+        assert(lhs < op.getOperation()->getNumOperands() && lhs >= 0 &&
+               rhs < op.getOperation()->getNumResults() && rhs >= 0);
+        return false;
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{Return the effective workload size for the operation.
```

Comment:
Is "workload size" a standard terminology? 
I had to look at the implementation to understand what this is because the description isn't enough here.